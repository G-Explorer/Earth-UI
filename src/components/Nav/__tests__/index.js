import React from 'react'
import { findDOMNode } from 'react-dom'
import TestUtils from 'react-dom/test-utils'
import { Nav, NavItem } from '../index'

// Object.defineProperty(window.location, 'href', {
//   writable: true,
//   value: '/'
// })

describe('Nav', () => {
  it('should defaultOpen works', () => {
    jsdom.reconfigure({
      href: '/'
    })
    const instance = TestUtils.renderIntoDocument(
      <Nav selectedId="1">
        <NavItem id="2" defaultOpen>
          <NavItem id="3" />
          <NavItem id="4" />
        </NavItem>
      </Nav>
    )
    expect(findDOMNode(instance).querySelector('li').className).toContain('open')
  })
})
