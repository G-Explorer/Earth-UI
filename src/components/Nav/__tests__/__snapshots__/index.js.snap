// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render correctly 1`] = `
<div
  className="true-nav"
>
  <ul>
    <SubNav
      defaultOpen={true}
      id="2"
      indent={24}
      key=".0"
    >
      <NavItemGroup>
        <NavItem
          id="3"
        />
        <NavItem
          id="4"
        />
      </NavItemGroup>
    </SubNav>
  </ul>
</div>
`;
